### PR TITLE
Field rations - Add inventory context menu action

### PR DESCRIPTION
**Pull request by shukari**
**When merged this pull request will:**
- ContextMenuOption for field_rations
- #include above comments
- ~~replace old ACE_medical_fnc_setDead~~
- localize LSTRING -> LLSTRING

### CODE REVIEW

File: addons/field_rations/XEH_preInit.sqf [hunk 1]
```diff
@@ -18,4 +18,11 @@ GVAR(waterSourceP3Ds) = _cache select 0;
 // List of refill action offsets corresponding to the p3ds in the array above
 GVAR(waterSourceOffsets) = _cache select 1;
 
+// ContextMenuOption
+["ACE_fieldRationsItems", ["CONTAINER"], "Eat/Drink", [], QPATHTOF(ui\icon_survival.paa), [{GVAR(enabled)}, {GVAR(enabled)}], {
```
Comment (PabstMirror) [inline addons/field_rations/XEH_preInit.sqf:22]: Maybe add a localize for "Eat/Drink"

Comment (mharis001) [inline addons/field_rations/XEH_preInit.sqf:22]: Move to `XEH_postInit.sqf`, add only if `field_rations` is enabled.

File: addons/field_rations/XEH_preInit.sqf [hunk 2]
```diff
@@ -18,4 +18,11 @@ GVAR(waterSourceP3Ds) = _cache select 0;
...
+        params ["_unit", "_container", "_item", "_slot", "_params"];
```
Comment (mharis001) [inline addons/field_rations/XEH_preInit.sqf:23]: Not all of these are used.

File: addons/field_rations/XEH_preInit.sqf [hunk 3]
```diff
@@ -18,4 +18,11 @@ GVAR(waterSourceP3Ds) = _cache select 0;
...
+        [objNull, _unit, _item] call acex_field_rations_fnc_consumeItem;
```
Comment (mharis001) [inline addons/field_rations/XEH_preInit.sqf:24]: `FUNC` macro.

**Comment by PabstMirror** [comment]
We are really close to next release, I've cherry picked the medical fix into #230

File: addons/field_rations/XEH_preInit.sqf (hunk 1 above)
Comment (shukari) [reply addons/field_rations/XEH_preInit.sqf:22]: the entry is only shown if it is enabled
i moved it to postInit, is there a special reason for that?

**Comment by shukari** [comment]
@mharis001 For my opinion it is better to have a base class for field_rations. So i only need to add one class.
Also the option is only shown if the settings is enabled and only usable if it has something in it

**Comment by mharis001** [comment]
Changing the items to a base class and limiting the action to it is the wrong decision at this point (consider `ACE_Banana` UBC and other 3rd party items).

**Comment by PabstMirror** [comment]
Probably fine to add action to `ACE_ItemCore` and make condition that item is food/drink

**Comment by shukari** [comment]
@mharis001 @PabstMirror 
use only ACE_ItemCore
show if enabled and field_rations items
enable if is full

**Comment by shukari** [comment]
need review @mharis001 @PabstMirror

File: addons/field_rations/XEH_postInit.sqf [hunk 4]
```diff
@@ -73,6 +73,27 @@ if !(hasInterface) exitWith {};
...
+                GVAR(enabled) && getNumber (_config >> _item >> QGVAR(consumeTime)) > 0
+            }
+        ],
+        {
+            params ["_unit", "", "_item"];
+            [objNull, _unit, _item] call FUNC(consumeItem);
+            false
+        },
+        false, _config
+    ] call CBA_fnc_addItemContextMenuOption;
```
Comment (mharis001) [inline addons/field_rations/XEH_postInit.sqf:95]:
```suggestion
    // Add inventory context menu option to consume items
    ["ACE_ItemCore", ["CONTAINER"], LSTRING(EatDrink), [], QPATHTOF(ui\icon_survival.paa),
        [
            {true},
            {
                params ["", "", "_item"];
                
                GVAR(enabled) && {
                    private _config = configFile >> "CfgWeapons" >> _item;                    
                    getNumber (_config >> QGVAR(thirstQuenched)) > 0
                    || {getNumber (_config >> QGVAR(hungerSatiated)) > 0}
                }
            }
        ],
        {
            params ["_unit", "", "_item"];
            [objNull, _unit, _item] call FUNC(consumeItem);
            false
        }
    ] call CBA_fnc_addItemContextMenuOption;
```

I don't think the consume time check or passing `configFile >> "CfgWeapons"` as a parameter are needed.

File: addons/field_rations/stringtable.xml [hunk 5]
```diff
@@ -880,5 +880,9 @@
             <Russian>Гуманитарный ИРП, предназначенный для питания людей во время кризисов и катастроф.</Russian>
             <Spanish>Una ración humanitaria destinada a servir como el suministro de alimentos de una persona durante las crisis</Spanish>
         </Key>
+        <Key ID="STR_ACEX_Field_Rations_ContextMenu_EatDrink">
```
Comment (mharis001) [inline addons/field_rations/stringtable.xml:883]: Can just be `EatDrink`. Also trivial, but move to top of file, probably under `MainInteractionText` string.

**Comment by shukari** [comment]
@mharis001 changes are made

**Pull request merged**